src/java/za/co/mip/ablduck/ABLDuck.java

```diff
@@ -0,0 +1,420 @@
+
+            List<String> hierarchy = result.getHierarchy();
+            Collections.reverse(hierarchy);
+            js.superclasses.addAll(hierarchy);
+
+            js.members.addAll(result.getInheritedmembers());
+
+            //Subclasses
+            for (Map.Entry<String, SourceJSObject> subclass:jsObjects.entrySet()) {
+                SourceJSObject subc = subclass.getValue();
+                if(subc.ext.equals(js.name)) 
+                    js.subclasses.add(subc.name);
+            }
+
+        }
+
+        //Write class js files out
+        for (Map.Entry<String, SourceJSObject> j:jsObjects.entrySet()) {
+            SourceJSObject js = j.getValue();
+
+            //Generate html
+            js.html = html.getClassHtml(jsObjects, js);
+    
+            File outputFile = new File(this.destDirOutput, js.name + ".js");
+            try (FileWriter file = new FileWriter(outputFile.toString())) {
+                file.write("Ext.data.JsonP." + js.name.replace(".", "_") + "(" + gson.toJson(js) + ");");
+            } catch (IOException ex) {
+                throw new BuildException(ex);
+            }
+        }
+
+        File dataFile = new File(this.destDir, "data.js");
+        try (FileWriter file = new FileWriter(dataFile.toString())) {
+            file.write("Docs = {\"data\":" + gson.toJson(dataJSObject) + "}");
+        } catch (IOException ex) {
+            throw new BuildException(ex);
+        }
+        
+    }
+
+    private HierarchyResult determineClassHierarchy(SourceJSObject curClass, HierarchyResult result) {
+        
+        String inherits = curClass.ext;
+
+        if (!"".equals(inherits)) {
+            result.addHierarchy(inherits);
+
+            SourceJSObject nextClass = jsObjects.get(inherits);
+                        
+            if (nextClass != null) {
+                for (MemberObject member : nextClass.members) {
+                    if (member.owner.equals(inherits) && (member.meta.isPrivate == null || !member.meta.isPrivate)) 
+                        result.addInheritedmember(member);
+                }
+            
+                determineClassHierarchy(nextClass, result);
+            }
+        }
+        return result;
+    }
+
+    private void extractTemplateDirectory(File outputDir) throws IOException {
+ 
+        File template = exportResource("ablduck.zip", outputDir);
+
+        unzip(template, outputDir);
+
+        if (!template.delete()) {
+            log("Unable to delete " + template.getAbsolutePath() + ", please delete this manually.", Project.MSG_INFO);
+        }
+        
+ 
+    }
+
+    public File exportResource(String resourceName, File outputDir) throws IOException {
+        
+        
+        File resource = new File(outputDir, resourceName);
+
+        try (
+            InputStream stream = ABLDuck.class.getResourceAsStream("resources/" + resourceName);
+            OutputStream resStreamOut = new FileOutputStream(resource.getAbsolutePath());

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this extraneous semicolon. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2959)